src/vos/vos_obj_index.c

```diff
@@ -240,7 +241,8 @@ vos_oi_find_alloc(struct vos_container *cont, daos_unit_oid_t oid,
 	d_iov_set(&val_iov, NULL, 0);
 	d_iov_set(&key_iov, &oid, sizeof(oid));
 
-	rc = dbtree_upsert(cont->vc_btr_hdl, BTR_PROBE_EQ, DAOS_INTENT_DEFAULT,
+	rc = dbtree_upsert(cont->vc_btr_hdl, BTR_PROBE_EQ,
```

Comment:
This has no effect for object/key trees.   All of the intent magic lives in the incarnation log after the punch model change.

Reply:
The caller offers the exact "intent" will make the logic to be more clear, as for how to use the "intent", it is the lower layer logic does, in spite of it is used by incarnation log or DTX. So at least, the patch clear the usage without breaking logic correctness.